### PR TITLE
DHCP options support

### PR DESCRIPTION
**Pull request by venkat-iblox**
- Will create a follow-up task to create acceptances for DHCP options in other IPAM objects

### CODE REVIEW

File: examples/resources/bloxone_dhcp_option_group/resource.tf [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+resource "bloxone_dhcp_option_group" "example_with_options" {
+  name     = "example_dhcp_option_group_with_options"
+  protocol = "ip6"
+
+  # Other Optional Fields
+  dhcp_options = [
+    {
+      type         = "boolean"
```
Comment (mathewab) [inline examples/resources/bloxone_dhcp_option_group/resource.tf:25]: type here should be `option` or `group`

File: examples/data-sources/bloxone_dhcp_option_codes/data-source.tf [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    name = "example-code"
+  }
+}
+
+# Get DHCP Option code/s by tag
+data "bloxone_dhcp_option_codes" "example_by_tag" {
+  tag_filters = {
+    location = "site1"
+  }
+}
```
Comment (mathewab) [inline examples/data-sources/bloxone_dhcp_option_codes/data-source.tf:13]: does option code have tags?

File: internal/service/ipam/api_option_code_data_source.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	client *bloxoneclient.APIClient
+}
+
+func (d *OptionCodeDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "dhcp_option_codes"
+}
+
+type IpamsvcOptionCodeModelWithFilter struct {
+	Filters    types.Map  `tfsdk:"filters"`
+	TagFilters types.Map  `tfsdk:"tag_filters"`
```
Comment (mathewab) [inline internal/service/ipam/api_option_code_data_source.go:36]: You can remove this as option code does not have tags

File: internal/service/ipam/api_option_code_resource.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+	client *bloxoneclient.APIClient
+}
+
+func (r *OptionCodeResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "dhcp_option_code"
+}
+
+func (r *OptionCodeResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "",
```
Comment (mathewab) [inline internal/service/ipam/api_option_code_resource.go:34]: Desciription?

File: internal/service/ipam/api_option_code_resource_test.go [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+	var resourceName = "bloxone_dhcp_option_code.test_comment"
+	var v ipam.IpamsvcOptionCode
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccOptionCodeComment("234", "basic_opt_code", "boolean", "boolean option code typ"),
```
Comment (mathewab) [inline internal/service/ipam/api_option_code_resource_test.go:139]: shouldn't really use typos to verify different `comment`. :)

File: examples/data-sources/bloxone_dhcp_option_codes/data-source.tf (hunk 2 above)
Comment (venkat-iblox) [reply examples/data-sources/bloxone_dhcp_option_codes/data-source.tf:13]: It doesn't. I will clean up the code

**Review by mathewab — APPROVED** [review]

**Pull request merged**